### PR TITLE
refactor(26237): Refactor the request infrastructure to the OpenAPI specs

### PR DESCRIPTION
**Pull request by vanch3d**
See https://hivemq.kanbanize.com/ctrl_board/57/cards/26237/details/

This PR is part of the bi-directional epic and refactors the query infrastructure to abide to the OpenAPI specs proposed in https://github.com/hivemq/hivemq-edge/pull/563

### Design 
The current implementation of the bi-directional features in the frontend is still based on a series of mocks (both data and process) to bypass the work-in-progress nature of the backend. 

To drive the alignment of the two parts of the engineering process, a proposal was made for the expected API endpoints. 

The PR refactors the bespoke mocks into a proper micro-service API mocking
- The OpenAPI specs are used to generate the stubs for the services and models 
- All the expected endpoints are covered by their relevant `React Query` hooks
- All the still-unsupported endpoints are covered by mocks, using `MSW` to intercept the requests and populate the returned body
- The internal MQTT Client, used for sampling topics and schema, has been "demoted" to `dev`-only and integrated with the relevant `MSW` handlers

Note that `MSW` needs to be activated for the mocks to be introduced. Change the following environment variable in the `env.local` files: 

```dotenv
VITE_FLAG_MOCK_SERVER=true
```

### CODE REVIEW

File: hivemq-edge/src/frontend/src/api/hooks/useProtocolAdapters/__handlers__/index.ts [hunk 1]
```diff
@@ -289,3 +297,30 @@ export const handlers = [
     return HttpResponse.json<ValuesTree>(mockDataPointOPCUA, { status: 200 })
   }),
 ]
+
+export const deviceHandlers = [
+  http.get('*/protocol-adapters/adapters/:adapterId/tags', ({ params }) => {
+    const { adapterId } = params
+    console.log('>', { adapterId })
```
Comment (h2xd) [inline hivemq-edge/src/frontend/src/api/hooks/useProtocolAdapters/__handlers__/index.ts:304]: watch out a console that you might not want

File: hivemq-edge/src/frontend/src/hooks/usePrivateMqttClient/usePrivateMqttClient.spec.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+import { SimpleWrapper as wrapper } from '@/__test-utils__/hooks/SimpleWrapper.tsx'
+import { usePrivateMqttClient } from '@/hooks/usePrivateMqttClient/usePrivateMqttClient.ts'
+import { PrivateMqttClientType } from '@/hooks/usePrivateMqttClient/type.ts'
+
+describe('usePrivateMqttClient', () => {
+  it('should load the data', async () => {
+    const { result } = renderHook(() => usePrivateMqttClient(), { wrapper })
+
+    expect(result.current).toStrictEqual<PrivateMqttClientType>({ state: undefined, actions: undefined })
```
Comment (h2xd) [inline hivemq-edge/src/frontend/src/hooks/usePrivateMqttClient/usePrivateMqttClient.spec.ts:12]: question: is it right that the data is empty? Does not match the title of the test

File: hivemq-edge/src/frontend/src/main.tsx [hunk 3]
```diff
@@ -23,7 +18,12 @@ if (body) {
 
 ReactDOM.createRoot(document.getElementById('root') as HTMLElement).render(
   <React.StrictMode>
-    <MainApp />
+    <ConditionalWrapper
+      condition={import.meta.env.VITE_FLAG_MOCK_SERVER === 'true'}
+      wrapper={(children) => <PrivateMqttClientProvider>{children}</PrivateMqttClientProvider>}
```
Comment (h2xd) [inline hivemq-edge/src/frontend/src/main.tsx:23]: Neat!

File: hivemq-edge/src/frontend/src/modules/Workspace/hooks/useGetFlowElements.spec.tsx [hunk 4]
```diff
@@ -23,6 +24,8 @@ vi.mock('@chakra-ui/react', async () => {
   return { ...actual, useTheme: vi.fn<[], Partial<WithCSSVar<Dict>>>(() => MOCK_THEME) }
 })
 
+//// 44-48,53-63,67-81,86-93
```
Comment (h2xd) [inline hivemq-edge/src/frontend/src/modules/Workspace/hooks/useGetFlowElements.spec.tsx:27]: question: what does that mean? seems like a debug comment to me

**Review by h2xd — APPROVED** [review]
LGTM! 🐊  Just found some small nits. Generally a really nice yet big PR :D

**Comment by h2xd** [comment]
Some unit tests seem to fail due to not matching objects, I've added my approval so you can move forward after they are turned green.

**Comment by h2xd** [comment]
Sorry miss clicked on the wrong button.

**Comment by vanch3d** [comment]
> Some unit tests seem to fail due to not matching objects, I've added my approval so you can move forward after they are turned green.

Tests are failing due to rebasing on a branch that need another significant refactoring (I know)

**Pull request merged**